crud/select.lua

```diff
@@ -247,9 +279,31 @@ function select_module.pairs(space_name, user_conditions, opts)
             error(string.format("Failed to get next object: %s", err))
         end
 
+        local space_format, err = utils.get_fields_format(
+                iter.space_format,
+                opts.fields
+        )
+
+        if err ~= nil then
+            return nil, err
+        end
+
+        if opts.fields then
+            tuple, err = utils.unflatten(tuple, iter.space_format)
+            if err ~= nil then
+                error(string.format("Failed to unflatten next object: %s", err))
+            end
+        end
```

Comment:
The same as below. After you fetch and sort tuples you should assign `nil`s to trailing fields that were merged.